tests/reftests/var-syntax.test

```diff
@@ -0,0 +1,208 @@
+N0REP0
+### <pkg:pkg++.1>
+opam-version: "2.0"
+### <pkg:pkg+.1>
+opam-version: "2.0"
+### <pkg:pkg+a.1>
+opam-version: "2.0"
+### <pkg:pkg++a.1>
+opam-version: "2.0"
+### <pkg:pkg1.1>
+opam-version: "2.0"
+### <pkg:pkg2.1>
+opam-version: "2.0"
+### <pkg:var-with-plus.1>
+opam-version: "2.0"
+build: [
+  ["echo" "%{?pkg++:installed:}%"]
+  ["echo" "%{?pkg+:installed:}%"]
+  ["echo" "%{?pkg+a:installed:}%"]
+  ["echo" "%{?pkg++a:installed:}%"]
+  ["echo" "%{?pkg1+pkg2:installed:}%"]
+  ["echo" "%{pkg1+pkg2:installed}%"]
+]
+depopts: [
+  "pkg++"
+  "pkg+"
+  "pkg+a"
+  "pkg++a"
+  "pkg1"
+  "pkg2"
+]
+### opam switch create test --empty
+### : Test the behaviour of %{?pkg+with+pluses:var:false}%
+### : See https://github.com/ocaml/opam/pull/5840
+### opam install --dry-run --verbose var-with-plus | sed-cmd echo
+The following actions will be simulated:
+=== install 1 package
+  - install var-with-plus 1
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+Processing  2/3: [var-with-plus: echo false]
++ echo "false" (CWD=${BASEDIR}/OPAM/test/.opam-switch/build/var-with-plus.1)
++ echo "false" (CWD=${BASEDIR}/OPAM/test/.opam-switch/build/var-with-plus.1)
++ echo "false" (CWD=${BASEDIR}/OPAM/test/.opam-switch/build/var-with-plus.1)
++ echo "false" (CWD=${BASEDIR}/OPAM/test/.opam-switch/build/var-with-plus.1)
++ echo "false" (CWD=${BASEDIR}/OPAM/test/.opam-switch/build/var-with-plus.1)
++ echo "false" (CWD=${BASEDIR}/OPAM/test/.opam-switch/build/var-with-plus.1)
```

Comment:
If you don't want to have all the echo for each run, it is possible to define a package for each `opam install pkg+`
```
### <pkg:var-with-plus.2>
opam-version: "2.0"
build: ["echo" "'?pkg++:installed:'" "%{?pkg++:installed:}%"]
depopts: [ "pkg++" ]
### opam install pkg++
### opam install --verbose var-with-plus.2 | sed-cmd echo
```
it'll also remove the unordered outputs

Reply:
Even defining each package at that moment :
```
### : End with ++ :
### <pkg:pkg++.1>
opam-version: "2.0"
### <pkg:var-with-plus.2>
opam-version: "2.0"
build: ["echo" "'?pkg++:installed:'" "%{?pkg++:installed:}%"]
depopts: [ "pkg++" ]
### opam install pkg++
### opam install --verbose var-with-plus.2 | sed-cmd echo

Reply:
My previous comment about labeling echoes is irrelevant if those one are implemented.